File: trains-service/internal/models/station.go
package models

// Station represents a train station entity with its associated fields.
type Station struct {
	ID     int64  `json:"id,omitempty"`
	Name   string `json:"name,omitempty"`
	City   string `json:"city,omitempty"`
	Region string `json:"region,omitempty"`
	Status Status `json:"status,omitempty"`
}

// UpdateStation represents the fields that can be updated for a station entity.
type UpdateStation struct {
	Name   string `json:"name,omitempty"`
	City   string `json:"city,omitempty"`
	Region string `json:"region,omitempty"`
	Status Status `json:"status,omitempty"`
}

File: frontend/src/modules/trains/types/station.ts
export interface Station {
    id: number;
    name: string;
    city: string;
    region: string;
    status: "active" | "inactive";
}

export interface CreateStationRequest {
    name: string;
    city: string;
    region: string;
    status: "active" | "inactive";
}

export interface UpdateStationRequest {
    name?: string;
    city?: string;
    region?: string;
    status?: "active" | "inactive";
}